game_engine/core/HaliteImpl.cpp

```diff
@@ -171,6 +171,13 @@ void HaliteImpl::run_game() {
     update_player_stats();
     game.replay.full_frames.back().add_end_state(game.store);
 
+	// Calculate carried_at_end
+	for (auto &[player_id, player] : game.store.players) {
+		for (const auto &[_entity_id, location] : player.entities) {
+			game.game_statistics.player_statistics[player_id.value].carried_at_end += game.store.get_entity(_entity_id).energy;
+		}
+	}
+
```

Comment:
If a bot is terminated (crashes/times out) should the halite carried at that point be counted for carried_at_end?

I could see arguments both ways and I'm not sure which is more useful for a general stat. 'Last turn alive' is a strict superset of 'game end', so the latter could be derived from the former if desired. But it would entail extra processing later if what everyone really wants is the game end number and not the last turn bot was alive.

The current implementation only counts it at the game end and will always have zero for bots that are terminated early.